deltacat/compute/compactor_v2/compaction_session.py

```diff
@@ -226,365 +135,69 @@ def _execute_compaction(
-    )
-    logger.info(
-        f"Deduped {total_dd_record_count} records and deleted {total_deleted_record_count} records..."
-    )
-
-    compaction_audit.set_input_records(total_input_records_count.item())
-
-    telemetry_time_merge = compaction_audit.save_step_stats(
-        CompactionSessionAuditInfo.MERGE_STEP_NAME,
+    build_uniform_deltas_result: tuple[
+        List[DeltaAnnotated], DeleteStrategy, List[DeleteFileEnvelope], Partition
+    ] = _build_uniform_deltas(
+        params, compaction_audit, input_deltas, delta_discovery_start
+    )
+    (
+        uniform_deltas,
+        delete_strategy,
+        delete_file_envelopes,
+        compacted_partition,
+    ) = build_uniform_deltas_result
+
+    # run merge
+    run_merge_result: tuple[
+        Optional[List[MergeResult]], np.int64, float, int, int, float
+    ] = _run_merge(
+        params,
+        uniform_deltas,
+        compacted_partition,
+        round_completion_info,
+        delete_strategy,
+        delete_file_envelopes,
+        compaction_audit,
+        previous_compacted_delta_manifest,
+        task_max_parallelism,
+    )
+    (
         merge_results,
-        merge_results_retrieved_at,
-        merge_invoke_end - merge_start,
-        merge_end - merge_start,
-    )
-
-    compaction_audit.set_records_deduped(total_dd_record_count.item())
-    compaction_audit.set_records_deleted(total_deleted_record_count.item())
-    mat_results = []
-    for merge_result in merge_results:
-        mat_results.extend(merge_result.materialize_results)
-
-    mat_results: List[MaterializeResult] = sorted(
-        mat_results, key=lambda m: m.task_index
-    )
-
-    hb_id_to_entry_indices_range = {}
-    file_index = 0
-    previous_task_index = -1
-
-    for mat_result in mat_results:
-        assert (
-            mat_result.pyarrow_write_result.files >= 1
-        ), "Atleast one file must be materialized"
-        assert (
-            mat_result.task_index != previous_task_index
-        ), f"Multiple materialize results found for a hash bucket: {mat_result.task_index}"
-
-        hb_id_to_entry_indices_range[str(mat_result.task_index)] = (
-            file_index,
-            file_index + mat_result.pyarrow_write_result.files,
-        )
-
-        file_index += mat_result.pyarrow_write_result.files
-        previous_task_index = mat_result.task_index
-
-    s3_utils.upload(
-        compaction_audit.audit_url,
-        str(json.dumps(compaction_audit)),
-        **params.s3_client_kwargs,
-    )
-
-    deltas = [m.delta for m in mat_results]
-
-    # Note: An appropriate last stream position must be set
-    # to avoid correctness issue.
-    merged_delta: Delta = Delta.merge_deltas(
-        deltas,
-        stream_position=params.last_stream_position_to_compact,
-    )
-
-    record_info_msg = (
+        total_hb_record_count,
+        telemetry_time_hb,
+        total_dd_record_count,
+        total_deleted_record_count,
+        telemetry_time_merge,
```

Comment:
Didn't we discuss update this in audit where this is calculated?

Reply:
Got it.. That explains the reasoning behind returning telemetry_time_merge but what about total_dd_record_count and other?